proto/rill/runtime/v1/api.proto

```diff
@@ -397,7 +404,12 @@ message ListFilesRequest {
 
 // Response message for RuntimeService.ListFiles
 message ListFilesResponse {
-  repeated string paths = 1;
+  repeated FileEntry files = 1;
+}
+
+message FileEntry {
```

Comment:
1. Add a docstring
2. If we're calling it `DirEntry` internally now, we should probably also call it that here to align names